src/lib/select/select.ts

```diff
@@ -630,8 +630,14 @@ export class MdSelect implements AfterContentInit, OnDestroy, OnInit, ControlVal
       wasSelected ? option.deselect() : option.select();
       this._sortValues();
     } else {
-      this._clearSelection(option);
-      this._selectionModel.select(option);
+      if (option.value == null) {
+        this._clearSelection();
+        this._onChange(option.value);
+        this.change.emit(new MdSelectChange(this, option.value));
```

Comment:
Rather than duplicating the change event logic from `propagateChanges`, seems like we could tweak `propagateChanges` to accept an optional value arg. I think it'd help make the conditional more readable.

```ts
private _onSelect(option) {
  if (this.multiple) {
      ...
  } else if (option.value == null) {
     this._clearSelection();
     this._propagateChanges(option.value);
  } else {
    this._clearSelection(option);
    this._selectionModel.select(option);
  }
...
}

private _propagateChanges(value?: any): void {
    let valueToEmit = Array.isArray(this.selected) ? this.selected.map(option => option.value)
        : (this.selected ? this.selected.value : value);
  ...
}

```